### PR TITLE
External Network: liqoctl network reset and disconnect

### PR DESCRIPTION
**Pull request by fra98**
# Description

This PR implements the liqoctl network reset and disconnect commands, and the REST Delete method for Configurations, PublicKeys, GatewayServers, GatewayClients

In addition:
- prevent running `liqoctl network connect` without issuing `liqoctl network init` first

### CODE REVIEW

File: cmd/liqoctl/cmd/network.go [hunk 1]
```diff
@@ -34,10 +34,14 @@ const liqoctlNetworkLongHelp = `Manage liqo networking.`
 
 const liqoctlNetworkInitLongHelp = `Initialize the liqo networking between two clusters.`
 
+const liqoctlNetworkResetLongHelp = `Reset the liqo networking between two clusters.`
```
Comment (cheina97) [inline cmd/liqoctl/cmd/network.go:37]: Can you add that it implicitly includes the disconnect logic?

File: cmd/liqoctl/cmd/network.go [hunk 2]
```diff
@@ -95,6 +101,21 @@ func newNetworkInitCommand(ctx context.Context, options *network.Options) *cobra
 	return cmd
 }
 
+func newNetworkResetCommand(ctx context.Context, options *network.Options) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "reset",
+		Short: "Reset the liqo networking between two clusters",
```
Comment (cheina97) [inline cmd/liqoctl/cmd/network.go:107]: Can you add that it implicitly includes the disconnect logic?

File: pkg/liqoctl/completion/completion.go [hunk 3]
```diff
@@ -303,6 +303,79 @@ func Gateway(ctx context.Context, f *factory.Factory, argsLimit int) FnType {
...
+	return common(ctx, f, argsLimit, retriever)
+}
+
+// GatewayServers returns a function to autocomplete GatewayServers names.
+func GatewayServers(ctx context.Context, f *factory.Factory, argsLimit int) FnType {
+	retriever := func(ctx context.Context, f *factory.Factory) ([]string, error) {
+		var gwServers networkingv1alpha1.GatewayServerList
+		if err := f.CRClient.List(ctx, &gwServers, client.InNamespace(f.Namespace)); err != nil {
+			return nil, err
+		}
```
Comment (cheina97) [inline pkg/liqoctl/completion/completion.go:319]: Can you use a function from package getters here?

File: pkg/liqoctl/completion/completion.go [hunk 4]
```diff
@@ -303,6 +303,79 @@ func Gateway(ctx context.Context, f *factory.Factory, argsLimit int) FnType {
...
+	return common(ctx, f, argsLimit, retriever)
+}
+
+// GatewayClients returns a function to autocomplete GatewayClients names.
+func GatewayClients(ctx context.Context, f *factory.Factory, argsLimit int) FnType {
+	retriever := func(ctx context.Context, f *factory.Factory) ([]string, error) {
+		var gwClients networkingv1alpha1.GatewayClientList
+		if err := f.CRClient.List(ctx, &gwClients, client.InNamespace(f.Namespace)); err != nil {
+			return nil, err
+		}
```
Comment (cheina97) [inline pkg/liqoctl/completion/completion.go:337]: Same here

File: pkg/liqoctl/completion/completion.go [hunk 5]
```diff
@@ -303,6 +303,79 @@ func Gateway(ctx context.Context, f *factory.Factory, argsLimit int) FnType {
...
+	return common(ctx, f, argsLimit, retriever)
+}
+
+// PublicKeys returns a function to autocomplete PublicKeys names.
+func PublicKeys(ctx context.Context, f *factory.Factory, argsLimit int) FnType {
+	retriever := func(ctx context.Context, f *factory.Factory) ([]string, error) {
+		var publicKeys networkingv1alpha1.PublicKeyList
+		if err := f.CRClient.List(ctx, &publicKeys, client.InNamespace(f.Namespace)); err != nil {
+			return nil, err
+		}
```
Comment (cheina97) [inline pkg/liqoctl/completion/completion.go:355]: Same here

File: pkg/liqoctl/completion/completion.go [hunk 6]
```diff
@@ -303,6 +303,79 @@ func Gateway(ctx context.Context, f *factory.Factory, argsLimit int) FnType {
...
+	return common(ctx, f, argsLimit, retriever)
+}
+
+// Configurations returns a function to autocomplete Configurations names.
+func Configurations(ctx context.Context, f *factory.Factory, argsLimit int) FnType {
+	retriever := func(ctx context.Context, f *factory.Factory) ([]string, error) {
+		var configurations networkingv1alpha1.ConfigurationList
+		if err := f.CRClient.List(ctx, &configurations, client.InNamespace(f.Namespace)); err != nil {
+			return nil, err
+		}
```
Comment (cheina97) [inline pkg/liqoctl/completion/completion.go:373]: Same here

File: pkg/liqoctl/network/cluster.go [hunk 7]
```diff
@@ -214,10 +268,67 @@ func (c *Cluster) EnsurePublicKey(ctx context.Context, remoteClusterIdentity *di
...
+func (c *Cluster) DeleteConfiguration(ctx context.Context, name string) error {
+	s := c.local.Printer.StartSpinner("Deleting network configuration")
+
+	conf := &networkingv1alpha1.Configuration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: c.local.Namespace,
+		},
+	}
+	if err := c.local.CRClient.Delete(ctx, conf); err != nil {
```
Comment (aleoli) [inline pkg/liqoctl/network/cluster.go:289]: should we ignore the not found error here? the same for the other deletes

**Pull request merged**